tests/integration/test_charm.py

```diff
@@ -173,5 +173,17 @@ async def test_build_and_deploy(ops_test: OpsTest):
 
     logger.info("Verifying history server has 1 app entry")
 
-    apps = json.loads(urllib.request.urlopen(f"http://{address}:18080/api/v1/applications").read())
+    for i in range(0, 5):
+        try:
+            apps = json.loads(
+                urllib.request.urlopen(f"http://{address}:18080/api/v1/applications").read()
+            )
+        except Exception:
+            apps = []
+
+        if len(apps) > 0:
+            break
+        else:
+            sleep(3)
```

Comment:
**question** Just out of curiosity, why do we need this for releasing?

Reply:
Not directly related but "release to charmhub" will run ci.yaml which should be robust.

This change is to prevent flakiness in the integration test due to possible lag between spark job run completion and movement of completed application logs to S3. Ideally should have gone in with the previous PR.